File: CS/WindowsApplication3/IsDayOffFunction.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.Data.Filtering;

namespace DXSample {
    public class IsDayOffFunction : ICustomFunctionOperatorBrowsable { 
        const string FunctionName = "IsDayOff";
        static readonly IsDayOffFunction instance = new IsDayOffFunction();

        public static void Register()
        {
            CriteriaOperator.RegisterCustomFunction(instance);
        }
        public static bool Unregister()
        {
            return CriteriaOperator.UnregisterCustomFunction(instance);
        }

        #region ICustomFunctionOperatorBrowsable Members

        public FunctionCategory Category
        {
            get {return FunctionCategory.DateTime; }
        }

        public string Description
        {
            get { return "IsDayOff"; }
        }

        public bool IsValidOperandCount(int count)
        {
           return count == 1;
        }

        public bool IsValidOperandType(int operandIndex, int operandCount, Type type)
        {
            return type == typeof(DateTime);
        }

        public int MaxOperandCount
        {
            get { return 1; }
        }

        public int MinOperandCount
        {
            get { return 1; }
        }

        #endregion

        #region ICustomFunctionOperator Members

        public object Evaluate(params object[] operands)
        {
            DateTime dt = Convert.ToDateTime(operands[0]);
            return dt.DayOfWeek == DayOfWeek.Sunday || dt.DayOfWeek == DayOfWeek.Saturday;
        }

        public string Name
        {
            get { return FunctionName; }
        }

        public Type ResultType(params Type[] operands)
        {
            return typeof(bool);
        }

        #endregion
    }
}

File: VB/WindowsApplication3/IsDayOffFunction.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports DevExpress.Data.Filtering

Namespace DXSample
	Public Class IsDayOffFunction
		Implements ICustomFunctionOperatorBrowsable
		Private Const FunctionName As String = "IsDayOff"
		Private Shared ReadOnly instance As New IsDayOffFunction()

		Public Shared Sub Register()
			CriteriaOperator.RegisterCustomFunction(instance)
		End Sub
		Public Shared Function Unregister() As Boolean
			Return CriteriaOperator.UnregisterCustomFunction(instance)
		End Function

		#Region "ICustomFunctionOperatorBrowsable Members"

		Public ReadOnly Property Category() As FunctionCategory Implements ICustomFunctionOperatorBrowsable.Category
			Get
				Return FunctionCategory.DateTime
			End Get
		End Property

		Public ReadOnly Property Description() As String Implements ICustomFunctionOperatorBrowsable.Description
			Get
				Return "IsDayOff"
			End Get
		End Property

		Public Function IsValidOperandCount(ByVal count As Integer) As Boolean Implements ICustomFunctionOperatorBrowsable.IsValidOperandCount
		   Return count = 1
		End Function

		Public Function IsValidOperandType(ByVal operandIndex As Integer, ByVal operandCount As Integer, ByVal type As Type) As Boolean Implements ICustomFunctionOperatorBrowsable.IsValidOperandType
			Return type Is GetType(DateTime)
		End Function

		Public ReadOnly Property MaxOperandCount() As Integer Implements ICustomFunctionOperatorBrowsable.MaxOperandCount
			Get
				Return 1
			End Get
		End Property

		Public ReadOnly Property MinOperandCount() As Integer Implements ICustomFunctionOperatorBrowsable.MinOperandCount
			Get
				Return 1
			End Get
		End Property

		#End Region

		#Region "ICustomFunctionOperator Members"

		Public Function Evaluate(ParamArray ByVal operands() As Object) As Object Implements DevExpress.Data.Filtering.ICustomFunctionOperator.Evaluate
			Dim dt As DateTime = Convert.ToDateTime(operands(0))
			Return dt.DayOfWeek = DayOfWeek.Sunday OrElse dt.DayOfWeek = DayOfWeek.Saturday
		End Function

		Public ReadOnly Property Name() As String Implements DevExpress.Data.Filtering.ICustomFunctionOperator.Name
			Get
				Return FunctionName
			End Get
		End Property

		Public Function ResultType(ParamArray ByVal operands() As Type) As Type Implements DevExpress.Data.Filtering.ICustomFunctionOperator.ResultType
			Return GetType(Boolean)
		End Function

		#End Region
	End Class
End Namespace